src/highlight.rs

```diff
@@ -0,0 +1,252 @@
+            theme: Box::new(
+                theme_set
+                    .themes
+                    .get(theme_name)
+                    .ok_or_else(|| anyhow!("Theme '{theme_name}' not found."))?
+                    .clone(),
+            ),
+        })
+    }
+
+    // FIXME: it's a bit odd that this is passed as an extra option.
+    // Maybe use a "specialized" enum instead that wraps it with the sytect options.
+    /// Get file type specific syntax for syntect highlighting.
+    ///
+    /// Returns None if the `DiffHighlighter::Simple` is used.
+    pub fn get_syntax(&self, path: &str) -> Option<&SyntaxReference> {
+        match self {
+            Self::Simple { .. } => None,
+            Self::Syntect { syntax_set, .. } => {
+                // TODO: probably better to use std::path?
+                let file_ext = path.rsplit('.').next().unwrap_or("");
+                Some(
+                    syntax_set
+                        .find_syntax_by_extension(file_ext)
+                        .unwrap_or_else(|| syntax_set.find_syntax_plain_text()),
+                )
+            }
+        }
+    }
+}
+
+pub fn highlight_hunk(
+    hunk: &Hunk,
+    hl: &DiffHighlighter,
+    syntax: Option<&SyntaxReference>,
+) -> String {
+    match hl {
+        DiffHighlighter::Simple {
+            color_added,
+            color_removed,
+        } => highlight_hunk_simple(hunk, *color_added, *color_removed),
+        DiffHighlighter::Syntect { syntax_set, theme } => {
+            highlight_hunk_syntect(hunk, syntax_set, theme, syntax.unwrap())
+        }
+    }
+}
+
+fn highlight_hunk_simple(
+    hunk: &Hunk,
+    color_added: crossterm::style::Color,
+    color_removed: crossterm::style::Color,
+) -> String {
+    let mut buf = String::new();
+    let color_added = style::SetForegroundColor(color_added).to_string();
+    let color_removed = style::SetForegroundColor(color_removed).to_string();
+
+    let (header_marker, header_content) = hunk.header();
+    buf.push_str(header_marker);
+    buf.push_str(header_content);
+    buf.push('\n');
+
+    for (line_type, line_content) in hunk.lines() {
+        match line_type {
+            DiffLineType::Unchanged => {
+                buf.push(' ');
+            }
+            DiffLineType::Added => {
+                buf.push_str(&color_added);
+                buf.push('+');
+            }
+            DiffLineType::Removed => {
+                buf.push_str(&color_removed);
+                buf.push('-');
+            }
+        }
+        buf.push_str(line_content);
+        buf.push('\n');
+    }
+
+    // workaround: remove trailing line break
+    buf.pop();
+    buf.push_str("\x1b[0m");
+    buf
+}
+
+fn highlight_hunk_syntect(
+    hunk: &Hunk,
+    syntax_set: &SyntaxSet,
+    theme: &Theme,
+    syntax: &SyntaxReference,
+) -> String {
+    // TODO: move somewhere else?
+    let marker_added = as_24_bit_terminal_escaped(&[(MARKER_ADDED_STYLE, MARKER_ADDED)], true);
```

Comment:
It is unclear to me why we use `as_24_bit_terminal_escaped` instead of the crossterm [Command API](https://docs.rs/crossterm/latest/crossterm/#command-api)